common/src/main/java/com/microsoft/identity/common/internal/net/HttpRequest.java

```diff
@@ -134,6 +135,9 @@ public static HttpResponse sendPost(final URL requestUrl, final Map<String, Stri
                         .putRequestIdHeader(requestHeaders == null ? null : requestHeaders.get(CLIENT_REQUEST_ID))
         );
 
+        final Map<String, String> telemetryHeaders = ServerTelemetry.getTelemetryHeaders();
```

Comment:
The network layer should not know about the telemetry layer, instead these headers should be passed in by the caller above

Reply:
Addressed